File: java/src/main/java/com/iteso/nintendo/attacks/ShadowPunch.java
package com.iteso.nintendo.attacks;

public class ShadowPunch implements Attack {
    static int damage;
    static String name = "Shadow Punch";

    @Override
    public String attackName() {
        return name;
    }

    @Override
    public int attackDamage() {
        return this.damage;
    }

    @Override
    public void setDamage(int damage) {
        this.damage = damage;
    }
}

File: python/ShadowPunch.py
class ShadowPunch:
    def __init__(self):
        self.damage = None
        self.name = 'Shadow Punch'

    def attackName(self):
        return self.name

    def attackDamage(self):
        return self.damage

    def setDamage(self, damage):
        self.damage = damage
